### PR TITLE
Fix zoom out UI scale

### PR DESCRIPTION
**Pull request by ajlende**
## What?

Fixes some UI issues in zoom-out mode such as bottom padding in Firefox and scale of the border in zoom out mode.

## Why?

Bug fix

## How?

Moves calculations into CSS where we can take advantage of the layout engine to simplify some calculations.

Also moved the iframe css to the iframe component instead of block-list.

## Testing Instructions

- See that the block borders in zoom out mode are the right width.
- See that there is a bottom margin for zoom out mode in Firefox.
- See that pages shorter than browser height show properly.

### CODE REVIEW

File: packages/edit-site/src/components/code-editor/style.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .edit-site-code-editor {
 	position: relative;
 	width: 100%;
-	min-height: 100%;
```
Comment (scruffian) [inline packages/edit-site/src/components/code-editor/style.scss]: This keeps the background filllng the screen.
Screenshot 2024-05-08 at 14 15 50

**Review by MaggieCabrera — APPROVED** [review]
LGTM pending on Ben's comments

File: packages/block-editor/src/components/iframe/content.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
+.block-editor-iframe__html {
+	transform-origin: top center;
+	transition: transform 0.3s;
+	@include reduce-motion("transition");
+}
+
+.block-editor-iframe__html.is-zoomed-out {
+	$scale: var(--wp-zoom-out-scale);
+	$frame-size: var(--wp-zoom-out-frame-size);
```
Comment (ellatrix) [inline packages/block-editor/src/components/iframe/content.scss]: Why did you move everything to CSS? We only need `--wp-zoom-out-scale`?
This is going to be problematic when I try to animate both scroll and scale in JS at the same time with the same function.

**Review by ellatrix — CHANGES_REQUESTED** [review]
Would love if we can only do the minimum needed changes

File: packages/block-editor/src/components/iframe/content.scss (hunk 2 above)
Comment (ellatrix) [reply packages/block-editor/src/components/iframe/content.scss]: Or I wonder if animating these variables will be the same in terms of performance. But still, I don't get why we make all of these variables if it's not needed.

Comment (ellatrix) [reply packages/block-editor/src/components/iframe/content.scss]: We might also want to prefix it with `wp-block-editor`, not sure

File: packages/block-editor/src/components/block-content-overlay/content.scss [hunk 3]
```diff
@@ -5,10 +5,9 @@
...
 
-	.block-editor-iframe__body.is-zoomed-out &::before {
-		// Unfortunately because of the vw unit, this is not always going to be exact
-		// When the scrollbar is visible, the frame exceeds the canvas by a few pixels.
-		width: calc(100vw);
-		left: calc(( 100% - 100vw ) / 2);
+	.block-editor-block-list__block.has-block-overlay::before {
+		left: 0;
+		right: 0;
+		width: auto;
```
Comment (ellatrix) [inline packages/block-editor/src/components/block-content-overlay/content.scss:11]: Why are we changing these rules?

File: packages/block-editor/src/components/iframe/content.scss (hunk 2 above)
Comment (ajlende) [reply packages/block-editor/src/components/iframe/content.scss]: > Or I wonder if animating these variables will be the same in terms of performance. But still, I don't get why we make all of these variables if it's not needed.

It was easier to read. I would expect the styling to be done in CSS. And it seemed like performance wouldn't be worse.

> This is going to be problematic when I try to animate both scroll and scale in JS at the same time with the same function.

Why do you need to animate them in JS? Can't the animations be done with CSS?

File: packages/block-editor/src/components/block-content-overlay/content.scss (hunk 3 above)
Comment (ajlende) [reply packages/block-editor/src/components/block-content-overlay/content.scss:11]: It provides exact sizing instead of using the hack that is sometimes wrong.

**Review by ellatrix — APPROVED** [review]

File: packages/block-editor/src/components/iframe/index.js [hunk 4]
```diff
@@ -221,44 +221,6 @@ function Iframe( {
...
-			const { documentElement } = body.ownerDocument;
-
-			body.classList.add( 'is-zoomed-out' );
-
-			documentElement.style.transform = `scale( ${ scale } )`;
-			documentElement.style.marginTop = `${ frameSize }px`;
-			// TODO: `marginBottom` doesn't work in Firefox. We need another way
-			// to do this.
-			documentElement.style.marginBottom = `${ bottomFrameSize }px`;
-			if ( iframeWindowInnerHeight > contentHeight * scale ) {
```
Comment (ellatrix) [inline packages/block-editor/src/components/iframe/index.js:238]: Let's see in a follow-up if this is still needed
